t/op/array.t

```diff
@@ -707,3 +707,13 @@ fresh_perl_is('my @x;$x[0] = 1;shift @x;$x[22] = 1;$x[25] = 1;','',
   {}, 'unshifting and growing an array initializes trailing elements');
 
 "We're included by lib/Tie/Array/std.t so we need to return something true";
+
+# GH #23447 - ensure that future optimizations don't break behaviour
+{
+    my @x = "a" .. "d"; sub f {} my $y = @x[2, 3, f()];
```

Comment:
For readability, I would prefer that this be broken into 3 statements each on its own line, *i.e.:*
```
    my @x = "a" .. "d";
    sub f {}
    my $y = @x[2, 3, f()];
```
At first I had trouble picking out the definition of `sub f {}`.